### PR TITLE
Fix TypeInfo.opEquals for some templates

### PR DESCRIPTION
**Pull request by rmanthorpe**
https://issues.dlang.org/show_bug.cgi?id=20579

### CODE REVIEW

**Comment by rmanthorpe** [comment]
I wish that were true, but `typeid` already doesn't work across DLL boundaries and that `opEquals` does nothing to help it. The reason being the cast to from `Object` to `TypeInfo` fails because it relies on `TypeInfo.opEquals` to do the cast. All the `opEquals` does right now is break `typeid` where there wasn't a problem before.

You're right the real solution is better `name` but your suggestion still isn't good enough, you need fully qualified names of all template parameters like `std.traits.fullyQualifiedName` does, or even better just directly comparing mangles.

File: src/object.d [hunk 1]
```diff
@@ -422,19 +420,20 @@ class TypeInfo
     @property immutable(void)* rtInfo() nothrow pure const @safe @nogc { return rtinfoHasPointers; } // better safe than sorry
 }
 
+// Bugzilla 20579
```
Comment (andralex) [inline src/object.d:423]:
```suggestion
// Bugzilla 20579: https://issues.dlang.org/show_bug.cgi?id=20579
```

File: src/object.d [hunk 2]
```diff
@@ -422,19 +420,20 @@ class TypeInfo
...
+@system unittest
```
Comment (andralex) [inline src/object.d:424]: Could this be `@safe`?

**Comment by kinke** [comment]
This is obsolete after #3527 now.

**Pull request closed without merging**